contracts/src/v0.8/dev/ArbitrumSequencerStatusRecorder.sol

```diff
@@ -0,0 +1,245 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {AddressAliasHelper} from "./vendor/arb-bridge-eth/v0.8.0-custom/contracts/libraries/AddressAliasHelper.sol";
+import {ForwarderInterface} from "./interfaces/ForwarderInterface.sol";
+import {AggregatorInterface} from "../interfaces/AggregatorInterface.sol";
+import {AggregatorV3Interface} from "../interfaces/AggregatorV3Interface.sol";
+import {AggregatorV2V3Interface} from "../interfaces/AggregatorV2V3Interface.sol";
+import {TypeAndVersionInterface} from "../interfaces/TypeAndVersionInterface.sol";
+import {FlagsInterface} from "./interfaces/FlagsInterface.sol";
+import {ArbitrumSequencerStatusRecorderInterface} from "./interfaces/ArbitrumSequencerStatusRecorderInterface.sol";
+import {SimpleReadAccessController} from "../SimpleReadAccessController.sol";
+import {ConfirmedOwner} from "../ConfirmedOwner.sol";
+
+/**
+ * @title ArbitrumSequencerStatusRecorder - L2 status history aggregator
+ * @notice L2 contract that receives status updates from a specific L1 address,
+ *  records a new answer if the status changed, and raises or lowers the flag on the
+ *   stored Flags contract.
+ */
+contract ArbitrumSequencerStatusRecorder is
+  AggregatorV2V3Interface,
+  ArbitrumSequencerStatusRecorderInterface,
+  TypeAndVersionInterface,
+  SimpleReadAccessController
+{
+  struct Round {
+    bool status;
+    uint64 timestamp;
+  }
+
+  event Initialized();
+  event L1OwnershipTransferred(address indexed from, address indexed to);
+
+  string private constant V3_NO_DATA_ERROR = "No data present";
+  /// @dev Follows: https://eips.ethereum.org/EIPS/eip-1967
+  address public constant FLAG_ARBITRUM_SEQ_OFFLINE =
+    address(bytes20(bytes32(uint256(keccak256("chainlink.flags.arbitrum-seq-offline")) - 1)));
```

Comment:
We should be able to generalise this contract for any L2 sequencer status by accepting this as a constructor argument - I'll create another PR to illustrate